src/vm/hints/hint_utils.rs

```diff
@@ -47,6 +46,42 @@ fn get_address_from_reference(
     None
 }
 
+fn compute_addr_from_one_offset(
+    relocatable: Relocatable,
+    hint_reference: &HintReference,
+) -> Option<MaybeRelocatable> {
+    if hint_reference.offset1.is_negative()
+        && relocatable.offset < hint_reference.offset1.abs() as usize
+    {
+        return None;
+    }
+    Some(MaybeRelocatable::from((
+        relocatable.segment_index,
+        (relocatable.offset as i32 + hint_reference.offset1) as usize,
+    )))
+}
+
+fn compute_addr_from_two_offsets(
+    relocatable: Relocatable,
+    hint_reference: &HintReference,
+) -> Option<MaybeRelocatable> {
+    let addr = if hint_reference.offset1.is_negative()
+        && relocatable.offset < hint_reference.offset1.abs() as usize
+    {
+        return None;
+    } else {
+        Relocatable {
+            segment_index: relocatable.segment_index,
+            offset: (relocatable.offset as i32 + hint_reference.offset1) as usize,
+        }
+    };
```

Comment:
In fact, it seems like it could simply be a single function adding both offsets. If one of them is `0` then the value will be the same adding it or not.